### PR TITLE
finished crossover code

### PR DESCRIPTION
**Pull request by roseypark19**

### CODE REVIEW

File: neuralnetwork/genome.js [hunk 1]
```diff
@@ -89,42 +89,54 @@ class Genome {
...
                 innovationMap[connection.innovation] = [];
             }
-            innovationMap[connection.innovation].push(connection);
+            innovationMap[connection.innovation].push({ ...connection });
         });
 
-        console.log(innovationMap)
+        let copiedConnections = [];
+
+        innovationMap.forEach(connectionList => {
```
Comment (apotafiy) [inline neuralnetwork/genome.js:153]: dont we want to choose the connections from the most fit genome? or are we starting assuming they are equal fitness

File: neuralnetwork/genome.js [hunk 2]
```diff
@@ -1,49 +1,68 @@
...
+
+    static NODE_TYPES = {
+        input: 0,
+        hidden: 1,
+        output: 2,
+    };
+
+    static INNOV_MAP = {};
+
+    static resetInnovations = () => { // this function must be called whenever a new set of agents is created
```
Comment (apotafiy) [inline neuralnetwork/genome.js:23]: why do we need to reset innovation numbers

File: neuralnetwork/genome.js [hunk 3]
```diff
@@ -25,8 +35,22 @@ class Genome {
...
     };
 
-    static getDefault = (numInputs, numHiddens, numOutputs, randomWeights = false) => {
+    static assignNodeId = (innovNum) => {
+        let id;
+        if (Genome.NODE_ID_MAP[innovNum] !== undefined) {
+            id = Genome.NODE_ID_MAP[innovNum];
+        } else {
+            id = Genome.NODE_ID++;
+            Genome.NODE_ID_MAP = id;
```
Comment (apotafiy) [inline neuralnetwork/genome.js]: isnt "Genome.NODE_ID_MAP" initialized as a list. but here "id" is an integer so we are assigning an integer to that variable?

Comment (roseypark19) [reply neuralnetwork/genome.js]: yep this is a typo. just pushed a fix

**Review by apotafiy — APPROVED** [review]

**Pull request merged**